v2/src/datadog.ts

```diff
@@ -30,6 +30,16 @@ import { LambdaFunction } from "./interfaces";
 
 const versionJson = require("../version.json");
 
+export class DatadogAspect implements IAspect {
+  constructor(private datadog: Datadog) {}
+
+  visit(node: IConstruct): void {
+    if (node.constructor.name === lambda.Function.name) {
```

Comment:
It seems that this condition might not instrument `lambda.SingletonFunction`, is this intentional?

Reply:
Ah that's a miss, good call

Reply:
That's what I seemed to run into.